### PR TITLE
Rewrite with React + TypeScript

### PR DESCRIPTION
**Pull request by graphemecluster**
New functions include:
- CodeMirror autocomplete, linting, fullscreen etc.
- Multiple schema editor instances
- UI controlling parameters by "選項"

### CODE REVIEW

**Comment by graphemecluster** [comment]
Note that files are fetched from [graphemecluster/qieyun-examples](https://github.com/graphemecluster/qieyun-examples) and [graphemecluster/qieyun-text-label](https://github.com/graphemecluster/qieyun-text-label) currently. These URLs should be changed later.

**Comment by graphemecluster** [comment]
There is still something to modify and discuss. I will put them in the Telegram group. Let’s discuss them before merging.

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html lang="zh-HK" dir="ltr">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="https://avatars3.githubusercontent.com/u/63484757" />
```
Comment (graphemecluster) [inline public/index.html:5]: Can the icon be loaded with this link in mainland?

Comment (ayaka14732) [reply public/index.html]: 大陸 load 唔到，不如用住 default icon 先，因為而家呢個 icon 都只係求其揾嘅，唔係最終版本。

**Comment by graphemecluster** [comment]
唔知點解將設定放喺 `.eslintrc.json` 會導致編譯發生問題，放落 `package.json` 就冇事

**Comment by ayaka14732** [comment]
其實嗰句 `"no-new-func": "off"` 唔應該設定為全局嘅

**Comment by graphemecluster** [comment]
（また余計なことをしちゃった💦……）
噉麻煩你幫我改返佢啦

**Comment by ayaka14732** [comment]
> 唔知點解將設定放喺 `.eslintrc.json` 會導致編譯發生問題，放落 `package.json` 就冇事

因為 `.eslintrc.json` 識別唔到 TypeScript 嘅嘢

**Review by ayaka14732 — APPROVED** [review]

**Pull request merged**